mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -361,13 +364,27 @@ class DotConverter final : public OpConversionPattern<migraphx::DotOp> {
     auto mop = rewriter.create<tosa::MatMulOp>(loc, newOutType, in_A, in_B);
 
     // Convert optional attributes
-    if (auto attr = op->getAttrOfType<BoolAttr>("xdlopsV2"))
-      mop->setAttr("xdlopsV2", attr);
-    if (auto attr = op->getAttrOfType<StringAttr>("perf_config"))
-      mop->setAttr("perf_config", attr);
+    if (auto attr = op->getAttr("xdlopsV2"))
+      mop->setAttr("xdlopsV2", attr.template cast<BoolAttr>());
```

Comment:
This will assert if given the wrong attr type (i.e. UnitAttr). I would keep the former since you are expecting these types anyway.

Reply:
> This will assert if given the wrong attr type (i.e. UnitAttr)

Not sure I catch your point. It was previously "assumed" to be `BoolAttr` as well. (So keeping the former way will still not allow a different attr type.) The only change I made is to move it from `getAttr` side to `setAttr` side. The reason why I need that is due to it is easier to this whole class is not templatized and it is easier for compiler to comply with C++ dot operator on the second line. 

Refer to: https://stackoverflow.com/questions/3505713/c-template-compilation-error-expected-primary-expression-before-token

Reply:
In the former, the `getAttrOfType` will return a null `attr` type if the op 1) does not contain the attr named or 2) contains the attr named but of a different type. Also, the `attr` will always have the specified type (ie. StringAttr).

In the latter, the 'getAttr' will return a generic Attribute which will fail the cast if it is a different type (UnitAttr vs. BoolAttr).

If you are having compile problems, you can explicitly assign type to `attr` accordingly (BoolAttr or StringAttr) in the conditional statement.

Reply:
If you prefer it not to silently ignore, you can use your method but check the type and error out accordingly before the explicit cast.

Reply:
Good point, thanks for elaborating it. Indeed spell out the type of the attribute in the conditional is a more defensive way to do it. Addressed in d7b8ad8